benchmark_marshaler_test.go

```diff
@@ -32,7 +32,7 @@ func Marshal(v interface{}) ([]byte, error) {
 	// Technique 1.
 	// Get type information and pointer from interface{} value without allocation.
 	typ, ptr := reflect.TypeAndPtrOf(v)
-	typeID := reflect.TypeID(typ)
+	typeID := reflect.TypeID(v)
```

Comment:
it's very funny here because I can't successfully run all benchmark and debug for a while because of this.